### PR TITLE
Config - Update benchmark naming to support annotations

### PR DESCRIPTION
**Pull request by abuccts**
__Description__

Update benchmark naming to support annotations.

__Major Revisions__
- Update name for `create_benchmark_context` in executor.
- Backward compatibility for model benchmarks using "_models" suffix.
- Update documents.

### CODE REVIEW

File: superbench/executor/executor.py [hunk 1]
```diff
@@ -224,10 +227,10 @@ def exec(self):
                         result = self.__exec_benchmark(context, log_suffix)
                         benchmark_results.append(result)
                 else:
-                    log_suffix = 'micro-benchmark {}'.format(benchmark_name)
+                    log_suffix = benchmark_name
                     logger.info('Executor is going to execute %s.', log_suffix)
                     context = BenchmarkRegistry.create_benchmark_context(
-                        benchmark_name,
+                        benchmark_real_name,
```
Comment (cp5555) [inline superbench/executor/executor.py:233]: Is it compatible with previous configuration?

**Review by cp5555 — APPROVED** [review]

File: superbench/executor/executor.py (hunk 1 above)
Comment (abuccts) [reply superbench/executor/executor.py:233]: yes, there's backward compatibility

**Review by cp5555 — APPROVED** [review]

File: docs/superbench-config.mdx [hunk 2]
```diff
@@ -208,7 +214,7 @@ ${benchmark_name}:
 #### Model-Benchmark
 
 ```yaml
-${name}_models:
+model-benchmarks:${annotation}:
```
Comment (guoshzhao) [inline docs/superbench-config.mdx:217]: Do we need to update the config files to align with the docs?

**Review by guoshzhao — APPROVED** [review]

File: docs/superbench-config.mdx (hunk 2 above)
Comment (abuccts) [reply docs/superbench-config.mdx:217]: it's backward competitive, current config files still work
we can update once when release next version

**Pull request merged**